### PR TITLE
(COV-936): Feat/WLS map data overlays

### PR DESCRIPTION
**Pull request by ocshawn**
[COV-936](https://occ-data.atlassian.net/browse/COV-936)
This is an internal branch of WLS code for developers to work on to get it ready to merge to production

### New Features
- Create Date slider on maps to view data changes over time
- Create multiple map layers to view other data

### CODE REVIEW

**Comment by ocshawn** [comment]
changes can be seen in QA https://qa-covid19.planx-pla.net/

File: src/Covid19Dashboard/IllinoisMapChart/index.jsx [hunk 1]
```diff
@@ -82,65 +93,118 @@ class IllinoisMapChart extends React.Component {
...
+    const startDate = new Date(curentDateRange.min);
+    const endDate = new Date(curentDateRange.max);
+
+    const dateDiff = msToDays(endDate - startDate);
+    let sliderValue;
+    let { sliderDate } = this.state;
+    if (sliderDate) {
+      // match date when switching between data sets
+
+      // TODO convert sliderDate to date object
```
Comment (ZakirG) [inline src/Covid19Dashboard/IllinoisMapChart/index.jsx:114]: Is this TODO still needed?

File: src/Covid19Dashboard/ControlPanel/index.jsx [hunk 2]
```diff
@@ -1,18 +1,48 @@
...
       <div className='control-panel'>
-        <h3>Confirmed cases</h3>
+        <h3>{this.props.legendTitle}</h3>
         <p>
-          Data source: <a href='https://systems.jhu.edu'>Johns Hopkins University CSSE</a>
+          Data source:&nbsp;
+          <a href={this.props.legendDataSource.link}>
+            {this.props.legendDataSource.title}
+          </a>
+          {this.props.legendTitle === 'Mobility Data'
```
Comment (ZakirG) [inline src/Covid19Dashboard/ControlPanel/index.jsx:25]: The string 'Mobility Data' -- will this only be set internally by your own code? Or do commons administrators need to take note of this special string when setting portal config?

File: src/Covid19Dashboard/IllinoisMapChart/index.jsx [hunk 3]
```diff
@@ -1,49 +1,31 @@
+/* eslint-disable react/sort-comp */
```
Comment (ZakirG) [inline src/Covid19Dashboard/IllinoisMapChart/index.jsx:1]: minor: is it necessary to disable this?

Comment (ocshawn) [reply src/Covid19Dashboard/IllinoisMapChart/index.jsx:1]: no its not, good catch i will remove and lint

File: src/Covid19Dashboard/ControlPanel/index.jsx [hunk 4]
```diff
@@ -1,18 +1,48 @@
...
+    const tooltipText2 = 'Additional data of interest to display in map tool-tip pop-up.  These data will not influence color gradients on the map.';
+
     return (
       <div className='control-panel'>
-        <h3>Confirmed cases</h3>
+        <h3>{this.props.legendTitle}</h3>
         <p>
-          Data source: <a href='https://systems.jhu.edu'>Johns Hopkins University CSSE</a>
+          Data source:&nbsp;
+          <a href={this.props.legendDataSource.link}>
```
Comment (ZakirG) [inline src/Covid19Dashboard/ControlPanel/index.jsx:22]: Can you add ` target="_blank"` to this anchor tag so that the link opens in a new tab?

**Review by ZakirG — APPROVED** [review]
Looks nice, works well. Good job ~

File: src/Covid19Dashboard/IllinoisMapChart/index.jsx (hunk 1 above)
Comment (ocshawn) [reply src/Covid19Dashboard/IllinoisMapChart/index.jsx:114]: ill remove the todo as it was merely a note for myself

File: src/Covid19Dashboard/ControlPanel/index.jsx (hunk 4 above)
Comment (ocshawn) [reply src/Covid19Dashboard/ControlPanel/index.jsx:22]: added

File: src/Covid19Dashboard/ControlPanel/index.jsx (hunk 2 above)
Comment (ocshawn) [reply src/Covid19Dashboard/ControlPanel/index.jsx:25]: currently set internally only

**Pull request merged**